components/LanguageSwitcher.tsx

```diff
@@ -27,7 +27,7 @@ export function LanguageSwitcher() {
   const languageOptions = React.useMemo(generateLanguageOptions, []);
   const defaultLanguage = languageOptions.find(language => language.value === intl.locale) || languageOptions[0];
   return (
-    <div className="relative">
+    <div className="relative max-w-[50%]">
```

Comment:
Generally controlling layout like this within a component should be avoided, better to control this from an element on the outside that tries to position the language switcher (which might be used differently in different places, sometimes taking the full width etc)